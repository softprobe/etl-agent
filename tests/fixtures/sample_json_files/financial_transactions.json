[
  {
    "transaction_id": "TXN-20240115-001",
    "account_id": "ACC-123456",
    "account_holder": {
      "name": "John Smith",
      "account_number": "1234567890",
      "routing_number": "021000021"
    },
    "transaction_date": "2024-01-15T14:30:25Z",
    "transaction_type": "debit",
    "amount": -45.67,
    "currency": "USD",
    "description": "STARBUCKS STORE #1234",
    "category": "food_dining",
    "merchant": {
      "name": "Starbucks",
      "mcc": "5814",
      "location": {
        "city": "San Francisco",
        "state": "CA",
        "country": "US"
      }
    },
    "balance_after": 1254.33,
    "metadata": {
      "channel": "pos",
      "card_present": true,
      "recurring": false
    }
  },
  {
    "transaction_id": "TXN-20240115-002", 
    "account_id": "ACC-789012",
    "account_holder": {
      "name": "Alice Johnson",
      "account_number": "9876543210",
      "routing_number": "021000021"
    },
    "transaction_date": "2024-01-15T16:45:10Z",
    "transaction_type": "credit",
    "amount": 2500.00,
    "currency": "USD",
    "description": "DIRECT DEPOSIT PAYROLL",
    "category": "income_salary",
    "merchant": {
      "name": "ACME Corp",
      "mcc": "0000"
    },
    "balance_after": 3750.50,
    "metadata": {
      "channel": "ach",
      "card_present": false,
      "recurring": true,
      "payroll_period": "biweekly"
    }
  },
  {
    "transaction_id": "TXN-20240115-003",
    "account_id": "ACC-123456", 
    "account_holder": {
      "name": "John Smith",
      "account_number": "1234567890",
      "routing_number": "021000021"
    },
    "transaction_date": "2024-01-15T19:22:33Z",
    "transaction_type": "debit",
    "amount": -125.99,
    "currency": "USD", 
    "description": "AMAZON.COM ORDER #123-456",
    "category": "shopping_online",
    "merchant": {
      "name": "Amazon",
      "mcc": "5942",
      "location": {
        "city": "Seattle",
        "state": "WA",
        "country": "US"
      }
    },
    "balance_after": 1128.34,
    "metadata": {
      "channel": "online",
      "card_present": false,
      "recurring": false,
      "order_id": "123-456-7890"
    }
  }
]