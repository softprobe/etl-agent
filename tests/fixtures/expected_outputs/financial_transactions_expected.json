{
  "business_entities": [
    {
      "name": "Transaction",
      "type": "financial_event",
      "primary_key": "transaction_id",
      "description": "Financial transaction record"
    },
    {
      "name": "Account",
      "type": "entity",
      "primary_key": "account_id", 
      "description": "Bank account information"
    },
    {
      "name": "AccountHolder",
      "type": "entity",
      "primary_key": null,
      "parent_entity": "Account",
      "description": "Account owner information"
    },
    {
      "name": "Merchant",
      "type": "entity", 
      "primary_key": null,
      "description": "Transaction merchant information"
    }
  ],
  "suggested_schema": {
    "transactions": {
      "transaction_id": {"type": "STRING", "nullable": false, "description": "Unique transaction identifier"},
      "account_id": {"type": "STRING", "nullable": false, "description": "Account identifier"},
      "transaction_date": {"type": "TIMESTAMP", "nullable": false, "description": "Transaction timestamp"},
      "transaction_type": {"type": "STRING", "nullable": false, "description": "Debit or credit"},
      "amount": {"type": "FLOAT", "nullable": false, "description": "Transaction amount"}, 
      "currency": {"type": "STRING", "nullable": false, "description": "Currency code"},
      "description": {"type": "STRING", "nullable": true, "description": "Transaction description"},
      "category": {"type": "STRING", "nullable": true, "description": "Transaction category"},
      "balance_after": {"type": "FLOAT", "nullable": true, "description": "Account balance after transaction"},
      "merchant_name": {"type": "STRING", "nullable": true, "description": "Merchant name"},
      "merchant_mcc": {"type": "STRING", "nullable": true, "description": "Merchant category code"},
      "merchant_city": {"type": "STRING", "nullable": true, "description": "Merchant city"},
      "merchant_state": {"type": "STRING", "nullable": true, "description": "Merchant state"},
      "merchant_country": {"type": "STRING", "nullable": true, "description": "Merchant country"},
      "channel": {"type": "STRING", "nullable": true, "description": "Transaction channel"},
      "card_present": {"type": "BOOLEAN", "nullable": true, "description": "Whether card was present"},
      "recurring": {"type": "BOOLEAN", "nullable": true, "description": "Whether transaction is recurring"}
    },
    "accounts": {
      "account_id": {"type": "STRING", "nullable": false, "description": "Account identifier"},
      "account_number": {"type": "STRING", "nullable": false, "description": "Account number"},
      "routing_number": {"type": "STRING", "nullable": false, "description": "Bank routing number"},
      "account_holder_name": {"type": "STRING", "nullable": false, "description": "Account holder name"}
    }
  },
  "confidence_scores": {
    "schema_detection": 0.93,
    "business_entity_detection": 0.90,
    "data_type_inference": 0.95, 
    "relationship_detection": 0.88
  },
  "potential_issues": [
    {
      "type": "sensitive_data",
      "description": "Account numbers and routing numbers are sensitive PII",
      "severity": "high",
      "suggested_solution": "Consider masking or hashing sensitive fields"
    },
    {
      "type": "nested_objects",
      "description": "Account holder and merchant data nested in transactions",
      "severity": "medium", 
      "suggested_solution": "Normalize into separate accounts and merchants tables"
    },
    {
      "type": "variable_metadata",
      "description": "Metadata fields vary by transaction type and channel",
      "severity": "low",
      "suggested_solution": "Create flexible metadata schema or separate metadata table"
    }
  ]
}